tests/conftest.py

```diff
@@ -17,56 +18,47 @@
     StatusSQL,
 )
 from pvsite_datamodel.write.datetime_intervals import get_or_else_create_datetime_interval
-from sqlalchemy import create_engine
-from sqlalchemy.orm import sessionmaker
 from testcontainers.postgres import PostgresContainer
 
 
 @pytest.fixture(scope="session", autouse=True)
-def engine():
+def database_connection():
     """Database engine, this includes the table creation."""
     with PostgresContainer("postgres:14.5") as postgres:
         url = postgres.get_connection_url()
         os.environ["OCF_PV_DB_URL"] = url
 
-        url = postgres.get_connection_url()
-        engine = create_engine(url)
-        Base.metadata.create_all(engine)
+        database_connection = DatabaseConnection(url)
 
-        yield engine
-
-        engine.dispose()
+        engine = database_connection.engine
 
+        Base.metadata.create_all(engine)
 
-@pytest.fixture(scope="session", autouse=True)
-def Session(engine):
-    """Returns a sessions making object.
+        yield database_connection
 
-    You can use `with Session() as session:`
-    """
-    return sessionmaker(bind=engine)
+        engine.dispose()
 
 
 @pytest.fixture()
-def db_session(Session):
```

Comment:
you can even make a sessions fixture, then most other fixtures just use that, so they dont have to worry about not closing the database connection e.t.c

Reply:
Isn't it what this fixture still is?

Reply:
oh sorry